lib/granite/translations.rb

```diff
@@ -2,18 +2,25 @@ module Granite
   module Translations
     extend ActiveSupport::Concern
 
-    def translate(*args)
-      I18n.translate(*self.class.scope_translation_args(args))
+    TranslationsWrapper = I18n
+
+    def translate(*args, **kwargs)
+      action_name = public_send(:action_name) if respond_to?(:action_name)
+
+      *i18n_args, i18n_options = self.class.scope_translation_args([*args, kwargs], action_name: action_name)
+      self.class.const_get('TranslationsWrapper').translate(*i18n_args, **i18n_options)
```

Comment:
I think this makes things more complicated than they have to be. Any reason we couldn't simply:
* Have `Granite::Translations` as you've defined, except instead of this call `super` with args.
* In `Granite::Action` include `Granite::Translations` and `Granite::I18nTranslations`, `Granite::I18nTranslations` would simply delegate `translate` to `I18n`
* In `Granite::Projector` include `Granite::Translations` and `Granite::I18nTranslations`
* In controller/view include `Granite::Translations`, they already have `t` and `translate` methods.